fluss-server/src/main/java/org/apache/fluss/server/coordinator/statemachine/TableBucketStateMachine.java

```diff
@@ -599,7 +633,18 @@ private Optional<ElectionResult> leaderForOffline(
                 new LeaderAndIsr(
                         leaderOpt.get(),
                         leaderAndIsr.leaderEpoch() + 1,
-                        leaderAndIsr.isr(),
+                        leaderAndIsr.isr().stream()
+                                .filter(
+                                        isr -> {
+                                            if (electionStrategy == CONTROLLED_SHUTDOWN_ELECTION) {
```

Comment:
All the `CONTROLLED_SHUTDOWN_ELECTION` strategy logic should happen in `controlledShutdownReplicaLeaderElection`.